frontend/src/components/Form.jsx

```diff
@@ -0,0 +1,204 @@
+import React, { useState } from "react";
+
+const _ = require('lodash');
+
+function Form(props) {
+    
+    const [formState, setFormField] = useState(() => setInitialState({}, props.fields));
+    const [imagePreview, setPreview] = useState(() => formState["image"]? formState["image"]["fileLink"] : "")
+
+    const imagestyle = {
+      height: "90px",
+      width: "90px",
+    };
+
+    function handleSubmit(e) {
+      
+      e.preventDefault();
+      props.callback(formState)
+    }
+
+    function setInitialState(newState, fields) {
+        _.forEach(fields, (field) => {
+            if (field.type === "TextString") {
+            newState[field.name] = {
+              hebrew: field.info.hebrew,
+              english: field.info.english,
+            };
+          } else {
+            newState[field.name] = field.info;
+          }
+        });
+        return newState
+    }
+
+    function createForm() {
+        
+        if (Object.keys(formState).length > 0) {
+            return _.map(props.fields, inputMaker);
+        } else {
+            return [];
+        }
+    }
+
+    function update(field) {
+      
+        if (field === "hebrew" || field === "english") {
```

Comment:
I don't understand the whole hebrew/english code - the lanugage can be saved in the context as a global configuration, and when you do api calls you'll pass that as well - but inner components don't need to know if it's hebrew or english - it doesn't matter to them

Reply:
I think we need to talk about this one